src/platform/unix/mod.rs

```diff
@@ -594,12 +596,18 @@ impl OsIpcOneShotServer {
             let fd = libc::socket(libc::AF_UNIX, SOCK_SEQPACKET, 0);
             let mut path: Vec<u8>;
             loop {
-                let path_string = CString::new(&b"/tmp/rust-ipc-socket.XXXXXX"[..]).unwrap();
+                let dir = TempDir::new().unwrap();
+                let dir_path = dir.path().join("rust-ipc-socket");
+                
+                //avoid deleting temp director automatically
+                let _tmp_path = dir.into_path();
+              
+                let mut _tmp_file = tempfile_in(&_tmp_path).unwrap();
```

Comment:
Are `_tmp_path` and `_tmp_file` needed? also `dir` doesn't live long enough. `TempDir` implements `Drop`, so it will be closed at the end of this funtion.

Reply:
yes, test run got stuck. If I remove _tmp_path and _tmp_file.  tempdir is getting dropped before socket bind operation. 
```
test platform::test::sender_transfer ... ok
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: UnixError(2)', libcore/result.rs:945:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
test platform::test::cross_process ... test platform::test::cross_process has been running for over 60 seconds
test platform::test::cross_process_sender_transfer ... test platform::test::cross_process_sender_transfer has been running for over 60 seconds
```
Same case when I remove _tmp_path. and use either `File::create` or `tempfile_in` using tempDir path. 

 ``` // let _tmp_path = dir.into_path();
    let mut _tmp_file = tempfile_in(&dir).unwrap(); ```

Reply:
> If I remove _tmp_path and _tmp_file. tempdir is getting dropped before socket bind operation.

Can't you add `dir` to the `OsIpcOneShotServer` so it doesn't get dropped.